recipe/build.sh

```diff
@@ -11,8 +11,22 @@ if test "${build_variant}" == "egl"; then
 fi
 
 if test "${CONDA_BUILD_CROSS_COMPILATION}" == "1"; then
-  # use native build tools
-  CC=$CC_FOR_BUILD CXX=$CXX_FOR_BUILD CFLAGS= CXXFLAGS= CPPFLAGS= LDFLAGS=${LDFLAGS//$PREFIX/$BUILD_PREFIX} \
+  if test "${target_platform}" == "osx-arm64"; then
+    # use native build tools
+    xCC=${CC_FOR_BUILD}
+    xCXX=${CXX_FOR_BUILD}
+    xCXXFLAGS= 
+    xCPPFLAGS= 
+    xLDFLAGS=${LDFLAGS//$PREFIX/$BUILD_PREFIX}
+  else
+    xCC=${CC}
+    xCXX=${CXX}
+    xCXXFLAGS=${CXXFLAGS}
+    xCPPFLAGS=${CPPFLAGS} 
+    xLDFLAGS=${LDFLAGS}
+  fi
+
+  CC=${xCC} CXX=${xCXX} CFLAGS=${xCFLAGS} CXXFLAGS=${xCXXFLAGS} CPPFLAGS=${xCPPFLAGS} LDFLAGS=${xLDFLAGS} \
      cmake -G "Ninja" -DCMAKE_PREFIX_PATH=$BUILD_PREFIX -DCMAKE_BUILD_TYPE=Release -B build-native .
```

Comment:
I dont see the point of precompiling build tools if its using the host target compiler anyway for the main build,
imho you could skip the whole "if test CONDA_BUILD_CROSS_COMPILATION" condition when its aarch64

Reply:
Does that mean to change the `if` to:
```
if test "${target_platform}" == "osx-arm64"; then
```
?

Reply:
I would rewrite the condition as follow: CONDA_BUILD_CROSS_COMPILATION==1 and CROSSCOMPILING_EMULATOR == ""

Reply:
But the enrichment of CMAKE_ARGS is needed for aarch64 too.
E.g. the build won't pass without `CMAKE_ARGS="${CMAKE_ARGS} -DQT_HOST_PATH=${BUILD_PREFIX}"`

Reply:
Please feel free to apply the changes your suggest. Maybe I don't fully understand you!